datasets/vivos/vivos.py

```diff
@@ -0,0 +1,124 @@
+    # You can create your own builder configuration class to store attribute, inheriting from datasets.BuilderConfig
+    # BUILDER_CONFIG_CLASS = MyBuilderConfig
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "speaker_id": datasets.Value("string"),
+                    "path": datasets.Value("string"),
+                    "sentence": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        dl_path = dl_manager.download_and_extract(_DATA_URL)
+        data_dir = os.path.join(dl_path, "vivos")
+        train_dir = os.path.join(data_dir, "train")
+        test_dir = os.path.join(data_dir, "test")
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(train_dir, "prompts.txt"),
+                    "path_to_clips": os.path.join(train_dir, "waves"),
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(test_dir, "prompts.txt"),
+                    "path_to_clips": os.path.join(test_dir, "waves"),
+                },
+            ),
+        ]
+
+    def _generate_examples(
+        self,
+        filepath,
+        path_to_clips,  # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+    ):
+        """Yields examples as (key, example) tuples."""
+        # This method handles input defined in _split_generators to yield (key, example) tuples from the dataset.
+        # The `key` is here for legacy reason (tfds) and is not important in itself.
+
+        with open(filepath, encoding="utf-8") as f:
+            lines = f.readlines()
+            for id_, row in enumerate(lines):

```

Comment:
```suggestion
            for id_, row in enumerate(f):
```

To avoid loading all the file content at once.